src/master-playlist-controller.js

```diff
@@ -523,21 +523,30 @@ export class MasterPlaylistController extends videojs.EventTarget {
    */
   fastQualityChange_() {
     const media = this.selectPlaylist();
+    const isPaused = this.tech_.paused();
 
     if (media === this.masterPlaylistLoader_.media()) {
       return;
     }
 
+    this.tech_.pause();
     this.masterPlaylistLoader_.media(media);
 
-    // delete all buffered data to allow an immediate quality switch, then seek
-    // in place to give the browser a kick to remove any cached frames from the
-    // previous rendition
+    // Delete all buffered data to allow an immediate quality switch, then seek
+    // forward slightly to give the browser a kick to remove any cached frames from the
+    // previous rendition (.04 seconds ahead is roughly the minimum that will accomplish this
+    // across a variety of content in IE and Edge)
+    // Edge/IE bug: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/14600375/
+    // Chrome bug: https://bugs.chromium.org/p/chromium/issues/detail?id=651904
     this.mainSegmentLoader_.resetEverything(() => {
       // Since this is not a typical seek, we avoid the seekTo method which can cause
       // segments from the previously enabled rendition to load before the new playlist
       // has finished loading
-      this.tech_.setCurrentTime(this.tech_.currentTime());
+      this.tech_.setCurrentTime(this.tech_.currentTime() + .04);
```

Comment:
Should we be seeking forwards for all browsers, or just those that are problematic?

Reply:
I was on the fence about whether it's better to minimize the scope of the forward seeking, or to omit any user agent checks and have uniform quality switch behavior across browsers. I'll defer to your judgement on this.

Reply:
I'd be interested to hear others' thoughts as well, but I think minimizing the scope may be better, just to limit potential impact, especially when it's only one set of browsers (IE/Edge).